monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ResultsFolderController.java

```diff
@@ -61,9 +70,22 @@
                     resultsFolder.getFiles(null, FileFilters.ANALYSIS_RESULT_SER.getExtension());
             for (final RepositoryFile file : files) {
                 final Map<String, String> map = new HashMap<>();
-                map.put("filename", file.getName());
-                map.put("repository_path", file.getQualifiedPath());
-                result.add(map);
+                final String name = file.getName();
+                if (searchedTimestamp != null) {
+                    // get the timestamp of the job
+                    final String timestampString = name.substring(name.lastIndexOf("-") + 1, name.indexOf("."));
+                    final Date jobDate = new Date(Long.valueOf(timestampString));
+                    if (jobDate.after(searchedTimestamp)) {
```

Comment:
These five lines should really all be in their own method, along with the parsing of the timestamp in the beginning.